calvinsys/cc_calvinsys.c

```diff
@@ -64,8 +64,13 @@ result_t calvinsys_register_capability(calvinsys_t *calvinsys, const char *name,
 	capability->handler = handler;
 	capability->state = state;
 
-	if (list_add_n(&calvinsys->capabilities, name, strlen(name), capability, sizeof(calvinsys_capability_t)) == CC_RESULT_SUCCESS)
+	if (list_add_n(&calvinsys->capabilities, name, strlen(name), capability, sizeof(calvinsys_capability_t)) == CC_RESULT_SUCCESS) {
+		list_t* head = calvinsys->capabilities;
+		while(head->next != NULL)
+			head = head->next;
+		handler->capability_name = head->id;
```

Comment:
Is the handler->capability_name used? I do not want to tie one handler to a specific capability, one handler can serve multiple capabilities. For example, a gpio handler can be used for the io.button and and io.motion_detection. The two capabilities uses the same handler.